### DISCUSSION TITLE
Date range filter for all widgets

### DISCUSSION THREAD

**Discussion by emincer** [filamentphp/filament#6310] [General]
Anyone know how we can do a date range filter for chart widgets? I can of course do it for each chart but that's not the best UX. Can anyone think of a way to have a drop down like last 7 days, last 30 days etc that can apply to all the dashboard widgets?

**Comment by danharrin** [comment]
The only way would be emitting a Livewire event when the filter is changed and listening on all the other components to reflect the change

**Reply by emincer** [reply]
This makes sense, thanks.  But is there a way to create the dropdown at the top level of the dashboard so it's the emitter?

**Reply by danharrin** [reply]
Using the form builder you could create a Select field
